torch_geometric/nn/conv/hetero_conv.py

```
@@ -0,0 +1,144 @@
+import torch
+import torch_geometric.nn as pyg_nn
+import torch_geometric.utils as pyg_utils
+import torch.nn as nn
+
+from torch import Tensor
+from torch._six import container_abcs
+from torch_geometric.nn.inits import reset
+from typing import (
+    List,
+    Dict,
+)
+
+
+class HeteroConv(torch.nn.Module):
+    r"""A "wrapper" layer designed for heterogeneous graph layers. It takes a
+    heterogeneous graph layer, such as :class:`torch_geometric.conv.hetero_sage_conv.HeteroSAGEConv`, 
+    at the initializing stage. 
+    .. note::
+        For more detailed use of :class:`HeteroConv`, TODO: example
+    """
+    def __init__(self, convs, aggr="add"):
+        super(HeteroConv, self).__init__()
+
+        assert isinstance(convs, container_abcs.Mapping)
+        self.convs = convs
+        self.modules = torch.nn.ModuleList(convs.values())
+
+        assert aggr in ["add", "mean", "max", "mul", "concat", None]
+        self.aggr = aggr
+
+    def reset_parameters(self):
+        r"""
+        """
+        for conv in self.convs.values():
+            reset(conv)
+
+    def forward(self, node_features, edge_indices, edge_features=None):
+        r"""The forward function for :class:`HeteroConv`.
+        Args:
+            node_features (Dict[str, Tensor]): A dictionary each key is node type and the corresponding
+                value is a node feature tensor.
+            edge_indices (Dict[str, Tensor]): A dictionary each key is message type and the corresponding
+                value is an `edge _ndex` tensor.
+            edge_features (Dict[str, Tensor]): A dictionary each key is edge type and the corresponding
+                value is an edge feature tensor. The default value is `None`.
+        """
+        
+        # node embedding computed from each message type
+        message_type_emb = {}
+        for message_key, message_type in edge_indices.items():
+            if message_key not in self.convs:
```

Do we need that? I would expect that such behavior should raise an error.

---

this is a potential functionality for partial / async message passing. E.g. in layer k, we can pass messages for a subset of message types, by only providing conv modules for which we perform message passing. Do you think it's needed?

---

I see. On the other-hand, this might lead to unexpected bugs. Hard for me to say which way is preferred.